elisp/intero.el

```diff
@@ -2015,15 +2015,22 @@ as (CALLBACK STATE REPLY)."
       (error "Intero process is not running: run M-x intero-restart to start it"))))
 
 (defun intero-network-call-sentinel (process event)
-  (with-current-buffer (process-buffer process)
+  (let* ((buf (process-buffer process))
+         (cleanup
+          (lambda ()
+            (delete-process process)
```

Comment:
I think that if you use `cl-flet` here instead, you wouldn't need to `funcall` the `cleanup` function. Looks reasonable otherwise.